llama-index-integrations/vector_stores/llama-index-vector-stores-mongodb/llama_index/vector_stores/mongodb/base.py

```diff
@@ -35,7 +35,7 @@ def _to_mongodb_filter(standard_filters: MetadataFilters) -> Dict:
     return filters
 
 
-class MongoDBAtlasVectorSearch(VectorStore):
+class MongoDBAtlasVectorSearch(BasePydanticVectorStore):
```

Comment:
This will fail pydantic when creating the class.

```
>>> import pydantic.v1 as pydantic
>>> from pydantic.v1 import BaseModel
>>> class Test(BaseModel):
...     def __init__(self):
...         self._val = 1
... 
>>> Test()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<stdin>", line 3, in __init__
  File "/home/loganm/.cache/pypoetry/virtualenvs/llama-index-4aB9_5sa-py3.10/lib/python3.10/site-packages/pydantic/v1/main.py", line 357, in __setattr__
    raise ValueError(f'"{self.__class__.__name__}" object has no field "{name}"')
ValueError: "Test" object has no field "_val"
>>> 
```

We need to add `PrivateAttr` fields for each `_X` class attribute. Check out other vector stores for an example